pldoc/static/js/ui.js

```diff
@@ -1,127 +1,133 @@
+
+                $('html, body').stop().animate({
+                    scrollTop: $(target).offset().top
+                }, 1000, 'swing', function() {
+                    Ui.sendFocus(target);
                 });
+            });
+        },
+
+        sendFocus: function(target) {
+            $(target).find('.pldoc-pattern-title:first')
+                     .attr('tabindex', '-1')
+                     .focus();
+        },
+
+        openNewWindow: function() {
+            $('a[rel="external"]').on('click', function(event) {
+                event.preventDefault();
+                window.open($(event.currentTarget).attr('href'));
+            });
+        },
+
+        // smoothscroll to target links
+        navigationHighlight: function() {
+            $('.pldoc-nav-patterns .pldoc-link').on('click', function(event) {
+                $('.pldoc-nav-patterns .pldoc-link').removeClass('is-current');
+                $(event.currentTarget).addClass('is-current');
+            });
+        },
+
+        // open external links in new windows
+        newWindowLink: function(event) {
+            event.preventDefault();
+            window.open($(event.currentTarget).attr('href'));
+        },
+
+        rgbaToHex: function(rgb) {
+            var rgbValue, hex;
+            if (typeof(rgb) !== 'undefined') {
+                rgbValue = rgb.match(/^rgba?[\s+]?\([\s+]?(\d+)[\s+]?,[\s+]?(\d+)[\s+]?,[\s+]?(\d+)[\s+]?/i);
+
+                hex = (rgbValue && rgbValue.length === 4) ? '#' +
+                ('0' + parseInt(rgbValue[1], 10).toString(16)).slice(-2) +
+                ('0' + parseInt(rgbValue[2], 10).toString(16)).slice(-2) +
+                ('0' + parseInt(rgbValue[3], 10).toString(16)).slice(-2) : '';
+
+                return hex;
+            } else {
+                return false;
             }
-        };
+        },
+
+        selectText: function(textToBeSelected) {
+            // this selects all the text in an element
+            // http://stackoverflow.com/questions/12243898/how-to-select-all-text-in-contenteditable-div
+            var doc = document,
+                range, selection,
+                element = textToBeSelected[0];
+
+            if (doc.body.createTextRange) {
+                range = doc.body.createTextRange();
+                range.moveToElementText(element);
+                range.select();
+            } else if (window.getSelection) {
+                selection = window.getSelection();
+                range = doc.createRange();
+                range.selectNodeContents(element);
+                selection.removeAllRanges();
+                selection.addRange(range);
+            }
+        },
+
+        listenForPaletteClick: function() {
+            $('.is-copyable').on('click', function() {
+                Ui.selectText($(event.currentTarget));
+            });
+        }
+    };
 
-        Ui.init();
-    });
+    Ui.init();
+});
```

Comment:
Nitpick: there should be a new line at the bottom of every file (edX code convention).